src/main/haskell/matching-logic/src/Kore/MatchingLogic/Signature.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE TypeFamilies #-}
+{-|
+Description: A type class for matching logic signatures
+
+Defines a type class for matching logic signatures.
+Data families 'Label' and 'Sort' indexed by the signature type
+give the labels and sorts of a signature. The class methods
+allow looking up the argument and result sorts of a label.
```

Comment:
I only plan to use this at the meta level, perhaps it could also be used on terms that are only at the object level. The logic rules do not currently describe how to handle terms that mix object and meta level.

This should be usable with `IndexedModule` in a similar way to `Kore.MatchingLogic.Signature.Simple`, or maybe even by directly using that module.